build.rs

```diff
@@ -20,6 +20,152 @@ fn build_config_name() -> String {
+        let exec_prefix: String = prefix.clone();
+        let lib_dir = format!(
+            "{exec_prefix}/lib/intel{arch}_{os}",
+            exec_prefix = exec_prefix,
+            arch = arch,
+            os = os
+        );
+        let omp_lib_dir = format!(
+            "{exec_prefix}/../compiler/lib/intel{arch}_{os}",
+            exec_prefix = exec_prefix,
+            arch = arch,
+            os = os
+        );
+        let include_dir = format!("{prefix}/include", prefix = prefix);
+
+        let mkl_root_path = PathBuf::from(mkl_root);
+        let lib_dir_path = PathBuf::from(lib_dir);
+        let omp_lib_dir_path = PathBuf::from(omp_lib_dir);
+        let include_dir_path = PathBuf::from(include_dir);
+
+        let mkl_root_str = mkl_root_path
+            .to_str()
+            .ok_or("Unable to convert 'mkl_root' to string")?;
+        let lib_dir_str = lib_dir_path
+            .to_str()
+            .ok_or("Unable to convert 'mkl_root' to string")?;
+        let omp_lib_dir_str = omp_lib_dir_path
+            .to_str()
+            .ok_or("Unable to convert 'mkl_root' to string")?;
+        let include_dir_str = include_dir_path
+            .to_str()
+            .ok_or("Unable to convert 'mkl_root' to string")?;
+
+        // Check that paths exist
+
+        if !mkl_root_path.exists() {
+            return Err(format!(
+                "The 'mkl_root' folder with path '{}' does not exist.",
+                mkl_root_str
+            ));
+        }
+
+        if !lib_dir_path.exists() {
+            return Err(format!(
+                "The 'lib_dir_path' folder with path '{}' does not exist.",
+                lib_dir_str
+            ));
+        }
+
+        if cfg!(feature = "openmp") {
+            if !omp_lib_dir_path.exists() {
+                return Err(format!(
+                    "The 'omp_lib_dir_path' folder with path '{}' does not exist.",
+                    omp_lib_dir_str
+                ));
+            }
+        }
+
+        if !include_dir_path.exists() {
+            return Err(format!(
+                "The 'include_dir_path' folder with path '{}' does not exist.",
+                include_dir_str
+            ));
+        }
+
+        Ok(MklDirectories {
+            mkl_root: mkl_root_str.into(),
+            lib_dir: lib_dir_str.into(),
+            omp_lib_dir: omp_lib_dir_str.into(),
+            include_dir: include_dir_str.into(),
+        })
+    }
+}
+
+fn lib_dirs_windows(mkl_dirs: &MklDirectories) -> Vec<String> {
+    if cfg!(feature = "openmp") {
+        vec![mkl_dirs.lib_dir.clone(), mkl_dirs.omp_lib_dir.clone()]
+    } else {
+        vec![mkl_dirs.lib_dir.clone()]
+    }
+}
+
+fn libs_windows() -> Vec<String> {
+    let libs_base = vec!["mkl_core_dll"];
+    let libs_seq = vec!["mkl_sequential_dll"];
+    let libs_omp = vec!["mkl_intel_thread_dll", "libiomp5md"];
+    let libs_lp64 = vec!["mkl_intel_lp64_dll"];
+    let libs_ilp64 = vec!["mkl_intel_ilp64_dll"];
```

Comment:
I don't fully understand why there is the `dll` suffix, but I guess it must be correct since you've evidently got it to compile!